state-chain/traits/src/lib.rs

```diff
@@ -446,6 +447,31 @@ where
 	}
 }
 
+pub trait CfeEventEmitterForCrypto<T: Chainflip, C: ChainCrypto> {
+	fn keygen_request(req: KeygenRequest<T::ValidatorId>);
+
+	fn signature_request(req: cfe_events::ThresholdSignatureRequest<T::ValidatorId, C>);
+
+	fn key_handover_request(_req: KeyHandoverRequest<T::ValidatorId, C>) {
+		assert!(!C::key_handover_is_required());
+	}
+}
+
+pub trait CfeEventEmitterT<T: Chainflip> {
```

Comment:
Why not something like `CfePeerRegistration`? 

We don't need to assume that the underlying implementation emits events.

Similarly for the other traits. Could be `CfeMultisigRequest`, `CfeBroadcastRequest`.

Reply:
Renamed traits in https://github.com/chainflip-io/chainflip-backend/pull/4382/commits/d2fdc660b486df0f599e7739c66fa1b3cb2cd522